server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -680,14 +681,23 @@ public void applyClusterState(ClusterChangedEvent event) {
         try {
             if (event.localNodeMaster()) {
                 // We don't remove old master when master flips anymore. So, we need to check for change in master
-                if (event.nodesRemoved() || event.previousState().nodes().isLocalNodeElectedMaster() == false) {
-                    processSnapshotsOnRemovedNodes(event);
+                final boolean newMaster = event.previousState().nodes().isLocalNodeElectedMaster() == false;
+                final SnapshotsInProgress snapshotsInProgress = event.state().custom(SnapshotsInProgress.TYPE);
+                if (snapshotsInProgress != null) {
```

Comment:
This check was nested in all the operations this condition now covers, much easier to understand that this method pretty much only takes effect when there's a `SnapshotsInProgress` in the state now imo.